### PR TITLE
chore: #79250 - Upgrade EDS to 0.21.0

### PR DESCRIPTION
**Pull request by pomfrida**
## Prerequisites

Please make sure you can check the following boxes:

-   [x] My code follows the code style of this project
-   [x] All new and existing tests passed
-   [x] Fix any eslint/prettier warnings/errors: npm run lint
-   [x] Remove unnecessary console.log (use logInfo logWarn if it's needed)
-   [x] Are variable and function names properly describing what it does?

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

-   [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
-   [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] I have updated the documentation accordingly
-   [ ] I have added tests to cover my changes

Description
Updating to EDS v0.21.0 (from 0.19.0). This includes a breaking change with `SingleSelect` and `MultiSelect` which are both deprecated and replaced by `Autocomplete` component. 

I updated the `TimePicker` Echo Component to be using `Autocomplete`:
As Autocomplete uses Downshift, it was updated with some extra padding on the input, which I had to fix for the placeholder to be displayed correctly. 

v0.21.0 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.21.0

v0.20.4 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.20.4

v0.20.3 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.20.3

v0.20.2 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.20.2

v0.20.1 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.20.1

v0.20.0 release notes:
https://github.com/equinor/design-system/releases/tag/eds-core-react%400.20.0

### How to test:
- npm run storybook
- check every component: see if you experience anything weird
  - focus on TimePicker - this is affected by the breaking changes (SingleSelect deprecated > Replaced by Autocomplete)
  
## Related PRs:
EchoFramework: https://github.com/equinor/EchoFramework/pull/297

### CODE REVIEW

File: src/__tests__/snapshots/elements/__snapshots__/SliderField.test.tsx.snap [hunk 1]
```diff
@@ -33,7 +37,7 @@ exports[`renders SliderField with labels correctly 1`] = `
       />
       <output
         className="Output__StyledOutput-sc-1dy945x-0 jaCOJW"
-        htmlFor="New slider with labels-thumb"
+        htmlFor="eds-thumb-82870"
```
Comment (pomfrida) [inline src/__tests__/snapshots/elements/__snapshots__/SliderField.test.tsx.snap:40]: Snapshot on PR/in GH actions is always failing on build. I have updated snapshot locally and it passes. So struggling with clearing this snapshot error atn

Comment (pomfrida) [reply src/__tests__/snapshots/elements/__snapshots__/SliderField.test.tsx.snap:40]: So everytime I update the test, it passes. :white_check_mark:  But when I run the tests again or push, the EDS ID changes and it fails again. So stuck in this loop

Comment (pomfrida) [reply src/__tests__/snapshots/elements/__snapshots__/SliderField.test.tsx.snap]: Adding `jest.spyOn(Math, 'random').mockReturnValue(0.12345)` to the test file solved it! 🥳

File: src/__tests__/snapshots/elements/SliderField.test.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-            />
-        )
-        .toJSON();
-    expect(tree).toMatchSnapshot();
-});
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { SliderField } from '../../../elements/sliderField/SliderField';
+
+jest.spyOn(Math, 'random').mockReturnValue(0.12345);
```
Comment (hkot1991) [inline src/__tests__/snapshots/elements/SliderField.test.tsx:5]: Can you add a comment as to why this is needed?

**Pull request merged**